services/streammanager/kafka/kafkamanager.go

```diff
@@ -328,14 +327,31 @@ func Produce(jsonData json.RawMessage, producer interface{}, destConfig interfac
 	message := prepareMessage(topic, userID, value, timestamp)
 
 	partition, offset, err := kafkaProducer.SendMessage(message)
+
+	var statusCode int
+	var returnMessage, errorMessage string
 	if err != nil {
+		// this will force retry in cases where kafka brokers went down
+		// after producer client had been already created
+
+		retryableErrors := []string{errOutOfBrokers, errCircuitBreakerIsOpen}
```

Comment:
declare this in kafkaerror.go instead of initializing it for every event?

Reply:
Yeah that makes sense.